core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2676,6 +2675,15 @@ public static void jsonArrayAggAdd(List list, Object element,
     }
   }
 
+  public static String jsonPretty(Object input) {
+    try {
+      return JSON_PATH_JSON_PROVIDER.getObjectMapper().writerWithDefaultPrettyPrinter()
+          .writeValueAsString(input);
+    } catch (Exception e) {
+      return null;
```

Comment:
This exception seems to be untrackable by user. For now I suggest to throw it using the tool class `org.apache.calcite.util.RESOURCE`. And once you are doing that, it will be better to add a test case to check the error message also.

Reply:
I have addressed the required changes. Thanks.